nav2_map_server/include/nav2_map_server/map_3d/map_io_3d.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2020 Shivam Pandey pandeyshivam2017robotics@gmail.com
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by shivam on 7/5/20.
+//
+
+#ifndef NAV2_MAP_SERVER__MAP_3D__MAP_IO_3D_HPP_
+#define NAV2_MAP_SERVER__MAP_3D__MAP_IO_3D_HPP_
+
+#include <string>
+#include <vector>
+
+#include "sensor_msgs/msg/point_cloud2.hpp"
+#include "geometry_msgs/msg/transform.hpp"
+#include "geometry_msgs/msg/vector3.hpp"
+#include "geometry_msgs/msg/quaternion.hpp"
+#include "geometry_msgs/msg/pose.hpp"
+
+namespace nav2_map_server
+{
+
+namespace map_3d
+{
+struct Origin
+{
+  std::vector<float> center;
+  std::vector<float> orientation;
+
+  void resize()
+  {
+    center.resize(3);
+    orientation.resize(4);
+  }
+};
+
+/**
+ * @brief parameters that will be populated while reading a YAML file
+ */
+struct LoadParameters
+{
+  std::string pcd_file_name;
+  Origin origin;
```

Comment:
Filling them with a for_loop kind structure won't be possible. This has been used in 
 saver_cli 
http://docs.ros.org/en/fuerte/api/geometry_msgs/html/msg/Point.html

Reply:
It is not clear what do you mean by for-loop? If it is used by map_saver_cli, this also should be changed there as well. It looks like we already discussed it previously, that all internal structures might be based on existing `Pose`. It is suitable, since contains the same fields as `Origin` has. Please use existing structures everywhere when it possible rather than re-invent existing things.